homeassistant/components/climate/daikin.py

```diff
@@ -136,11 +145,17 @@ def get(self, key):
         elif key == ATTR_OPERATION_MODE:
             # Daikin can return also internal states auto-1 or auto-7
             # and we need to translate them as AUTO
-            value = re.sub(
+            tmp_value = re.sub(
                 '[^a-z]',
                 '',
                 self._api.device.represent(daikin_attr)[1]
-            ).title()
+            )
+
+            daikin_op_mode = DAIKIN_TO_HA_STATE[tmp_value]
+            if daikin_op_mode is not None:
+                value = DAIKIN_TO_HA_STATE.get(daikin_op_mode)
```

Comment:
How can this work? We already have the value from the dict in `daikin_op_mode`. We shouldn't need to get it again.

Reply:
I think that I can make this part a little bit better, as soon as I can I will do that. The if statement is definetly too much. But the dictionary DAIKIN_TO_HA_STATE is necessary to map the operation_mode from the AC into the accepted one in HA climate component (there are some mismatch, like ‘heat’ vs ‘hot’). That’s way in the previous version the localization wasn’t working.

Reply:
Yes, but you're using the same dictionary twice. That looks like a bug. We should only need to do one lookup.

Reply:
Dictionary is what we always use for translating modes between home assistant and device. You still didn't explain what you want to do with this part of the code. Do we want to end up with Daikin mode or home assistant mode, after the translation?

If it's home assistant mode, we should only use `DAIKIN_TO_HA_STATE` dictionary one time, not twice in a row.

Reply:
This is how I would expect us to do it:
```py
daikin_mode = re.sub(
    '[^a-z]', '', self._api.device.represent(daikin_attr)[1])

ha_mode = DAIKIN_TO_HA_STATE.get(daikin_mode)
value = ha_mode
```